### PR TITLE
BUG #1886: Polar projection does not change pole.

### PR DESCRIPTION
**Pull request by danlipsa**
Polar projection should use:
South Pole for y1 < y2
North Pole for y1 > y2.

### CODE REVIEW

**Comment by danlipsa** [comment]
Baselines in
https://github.com/UV-CDAT/uvcdat-testdata/pull/120

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 Please review. See also the issues observed while debugging this:
https://github.com/UV-CDAT/uvcdat/issues/1901
https://github.com/UV-CDAT/uvcdat/issues/1903
https://github.com/UV-CDAT/uvcdat/issues/1905

**Comment by danlipsa** [comment]
The following tests still fail on master and on this branch:
	 43 - vcs_read_old_scr_2 (Failed)
	423 - vcs_test_continents (Failed)
	561 - diags_test_02 (Failed)
	562 - diags_test_03 (Failed)
	563 - diags_test_04 (Failed)
	564 - diags_test_41 (Failed)
	565 - diags_test_05 (Failed)
	566 - diags_test_06 (Failed)
	567 - diags_test_07 (Failed)
	568 - diags_test_08 (Failed)
	569 - diags_test_09 (Failed)
	570 - diags_test_10 (Failed)
	571 - diags_test_11 (Failed)
	572 - diags_test_12 (Failed)
	573 - diags_test_13 (Failed)
	574 - diags_test_15 (Failed)

**Comment by aashish24** [comment]
@danlipsa why there is a hole in test_vcs_basic_isofill_-3_proj_gmflip_NH_via_gm.png?

File: Packages/vcs/vcs/vcs2vtk.py [hunk 1]
```diff
@@ -251,11 +251,18 @@ def genGridOnPoints(data1, gm, deep=True, grid=None, geo=None,
     if geo is None:
         bounds = pts.GetBounds()
         xm, xM, ym, yM = [bounds[0], bounds[1], bounds[2], bounds[3]]
-        # We don't use the zooming feature (gm.datawc) for geographic
-        # projections. We use wrapped coordinates for doing the projection
+        # We use zooming feature (gm.datawc) for linear and ploar projections.
+        # We use wrapped coordinates for doing the projection
```
Comment (aashish24) [inline Packages/vcs/vcs/vcs2vtk.py]: this sentence does not read well --> such that parameters such that central meridian are set correctly

@danlipsa could you please clean it up?

File: Packages/vcs/vcs/vcs2vtk.py [hunk 2]
```diff
@@ -251,11 +251,18 @@ def genGridOnPoints(data1, gm, deep=True, grid=None, geo=None,
...
         # such that parameters such that central meridian are set correctly
+        if (gm.g_name == 'Gfm'):
```
Comment (aashish24) [inline Packages/vcs/vcs/vcs2vtk.py]: What does Gfm stands for?

File: Packages/vcs/vcs/template.py [hunk 3]
```diff
@@ -1595,7 +1595,7 @@ def plot(self, x, slab, gm, bg=0, min=None,
             Y = slab.getAxis(-2)
 
         wc2 = vcs.utils.getworldcoordinates(gm, X, Y)
-        wc2 = kargs.get("dataset_bounds", wc2)
+        wc2 = kargs.get("plotting_dataset_bounds", wc2)
```
Comment (aashish24) [inline Packages/vcs/vcs/template.py]: where is plotting_dataset_bounds defined?

**Comment by aashish24** [comment]
@danlipsa I have some questions but I see the pattern for meshfill which seems reasonable to me but I will have detailed look after I get the answers.

File: Packages/vcs/vcs/vcs2vtk.py (hunk 2 above)
Comment (danlipsa) [inline Packages/vcs/vcs/vcs2vtk.py]: meshfill

File: Packages/vcs/vcs/template.py (hunk 3 above)
Comment (danlipsa) [inline Packages/vcs/vcs/template.py]: VTKPlots.py

File: Packages/vcs/vcs/vcs2vtk.py (hunk 1 above)
Comment (danlipsa) [inline Packages/vcs/vcs/vcs2vtk.py]: Sure, I'll cleanup the comment. The command for  test_vcs_basic_isofill_-3_proj_gmflip_NH_via_gm.png is

"/home/danlipsa/src/uvcdat/testing/vcs/test_vcs_basic_gms.py" "--gm_type=isofill" "--projection=-3" "--lat1=90" "--lat2=0" "--range_via_gm" "--gm_flips_lat_range" "--source=/home/danlipsa/build/uvcdat/uvcdat-testdata/baselines/vcs/test_vcs_basic_isofill_-3_proj_gmflip_NH_via_gm.png"

so
(90, 0) flipped -> (0, 90).
which means view from the South.
The hole is (-90, 0). Note the NH which is appended to the file is wrong: https://github.com/UV-CDAT/uvcdat/issues/1903

**Comment by doutriaux1** [comment]
@danlipsa meshfill will be fixed in a separate PR correct?

**Comment by danlipsa** [comment]
@doutriaux1 Yes, we can definitely work on fixing the meshfill. https://github.com/UV-CDAT/uvcdat/issues/1905

**Comment by doutriaux1** [comment]
Ok I'll test the updated PR and merge in then.

**Comment by danlipsa** [comment]
@doutriaux1 Note meshfill might be trickier to get working as zooming does not work correctly for it:
https://github.com/UV-CDAT/uvcdat/issues/1912

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 Do I merge this in or any of you want to merge it?

**Pull request merged**